libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -152,6 +152,54 @@ string YulUtilFunctions::storeLiteralInMemoryFunction(string const& _literal)
 	});
 }
 
+string YulUtilFunctions::copyLiteralToStorageFunction(string const& _literal)
+{
+	string functionName = "copy_literal_to_storage_" + util::toHex(util::keccak256(_literal).asBytes());
+
+	return m_functionCollector.createFunction(functionName, [&]() {
+		Whiskers templ(R"(
+			function <functionName>(slot) {
+				<body>
+			}
+		)");
+		templ("functionName", functionName);
+
+		if (_literal.size() >= 32)
+		{
+			size_t words = (_literal.length() + 31) / 32;
+			vector<map<string, string>> wordParams(words);
+			for (size_t i = 0; i < words; ++i)
+			{
+				wordParams[i]["offset"] = to_string(i);
+				wordParams[i]["wordValue"] = formatAsStringOrNumber(_literal.substr(32 * i, 32));
+			}
+			templ("body", Whiskers(R"(
+					<resizeArray>(slot, <length>)
+					let dstPtr := <dataLocation>(slot)
+					<#word>
+						sstore(add(dstPtr, <offset>), <wordValue>)
+					</word>
+				)")
+				("resizeArray", resizeDynamicByteArrayFunction(*TypeProvider::bytesStorage()))
+				("dataLocation", arrayDataAreaFunction(*TypeProvider::bytesStorage()))
+				("word", wordParams)
+				("length", to_string(_literal.size()))
+				.render());
+		}
+		else
+			templ("body", Whiskers(R"(
+					<resizeArray>(slot, <length>)
+					sstore(slot, add(<wordValue>, mul(2, <length>)))
```

Comment:
So this ends up being (on default settings for optimizer):
```
function copy_literal_to_storage_4e03657aea45a94fc7d47ba826c8d667c0d1e6e33a64a036ec44f58fa12d6c45(slot)
{
    resize_array_t_bytes_storage_ptr(slot, 3)
    sstore(slot, add("abc", mul(2, 3)))
}
```
Also, this resize function has two cases where it calls different function for increasing and decreasing size where the actual write of length is happening. Optimizing `sstore`s depends on if function would be inlined or not. I think it was mentioned in #10479 

What are the chances for optimizer to optimize this out soon? @hrkrshnn @chriseth @ekpyron 
If it is unlikely, maybe better doing some inlined checks here and calling some kind of `clearStorageRange` function instead of full resize call.

`mul(2, 3)` is obviously compile time constant and I have changed it now so we set it to 6 in compile time...